src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FallThroughCheck.java

```diff
@@ -273,7 +273,7 @@ private boolean checkIf(final DetailAST ast, boolean useBreak,
             isTerminated = isTerminated(elseStmt.getFirstChild(),
                 useBreak, useContinue);
         }
-        else if (elseStmt == null) {
+        else {
             isTerminated = false;
         }
```

Comment:
Top condition has 2 parts which make the if on the else unnecessary.
1) Top condition has `elseStmt != null`, If that is ever false, obviously the `elseStmt == null` is then always `true` no matter what.
2) Top condition has `isTerminated`. If that is ever false and goes to the else, the else statement just resets `isTerminated` back to `false` so it doesn't matter if we check the condition or not.

Since I can prove the `else if` isn't needed by logic, I assume no regression is necessary.